rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,233 @@
+
+# Adoption strategy
+
+This proposed work will be implemented as a `BREAKING CHANGE`. In terms of how consuming projects will upgrade Carbon, we will need to document the upgrade process and link to it in the release notes. If a project has used an input as uncontrolled, they will need to update their codebase to make this controlled instead. This will involve making sure that the value is stored using React's [useState](https://reactjs.org/docs/hooks-state.html) hook to retain the value and making sure this is managed using an `onChange` function. We may have to provide additional support in this scenario to provide examples of how they can achieve the same functionality using this approach.
+
+We also already provide various codemods in our [Carbon Codemod](https://github.com/Sage/carbon-codemod) repo that could be used by our consumers to assist with the move to a controlled component. Examples of this are:
+- A [rename-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop) codemod that consumers could use to change a `defaultValue` value prop to `value`.
+- A [add-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/add-prop) codemod that could be used to add to a `value` prop.
+- A [remove-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/remove-prop) codemod that could be used to remove a `defaultValue` prop.
+- A [replace-prop-value](https://github.com/Sage/carbon-codemod/tree/master/transforms/replace-prop-value) codemod that changes the value of a prop.
+
+There is also support for writing controlled components in React's documentation. This can be found [here](https://reactjs.org/docs/forms.html#controlled-components).
+
+# How we teach this
+
+This RFC will be communicated in the Sage `#carbon` and `#carbon-announcement` slack channels, where we will ask our consumers to feedback on the RFC. Beyond that, it will be visible in the release notes when consumers upgrade. This RFC document will reside in the repository for future reference.
+
+We will be able to support consumers of Carbon in the `#carbon` slack channel too. 
+
+# Documentation
+
+As part of this work we will make sure that the documentation reflects what we will support. We will update the component description to include that we specifically only support controlled usage of these components. We will also ensure Storybook examples are updated where any uncontrolled behaviours are demonstrated. This will involve removing any uncontrolled examples and ensuring we have the controlled examples.
```

Comment:
`reflects what we will support` - I don't think the "will" is necessary, as this is talking about a future time after the support for uncontrolled components is removed.

Also the 2 sentences about updating Storybook documentation would be clearer as just 1 imo: `We will also ensure Storybook examples are updated to remove or update any that use uncontrolled components.`